### PR TITLE
Monoid complete

### PR DESCRIPTION
**Pull request by expede**
So, I figured that committing to `master` all the time was _probably a bad idea_. Just rounding out the `Monoid` module. It's pretty small, but was good to get the document generator working the way I like it, and whatnot.

### CODE REVIEW

File: lib/witchcraft.ex [hunk 1]
```diff
@@ -1,2 +1,25 @@
 defmodule Witchcraft do
+  import Witchcraft.Monoid
+  import Witchcraft.Monoid.Functions
+  import Witchcraft.Monoid.Properties
+
+  # import Witchcraft.Functor
```
Comment (expede) [inline lib/witchcraft.ex]: These are expected to be commented out, until the functionality is complete.

**Comment by expede** [comment]
@jenncoop, this totally works. It's more to 1) get you used to reading Elixir / running the tests, and 2) lower the bus factor. If you could [read the docs for the Monoid protocol](http://www.robotoverlord.io/witchcraft/Witchcraft.Monoid.html) (and the [Functions](http://www.robotoverlord.io/witchcraft/Witchcraft.Monoid.Functions.html) and [Properties](http://www.robotoverlord.io/witchcraft/Witchcraft.Monoid.Properties.html) modules) and let me know if they're comprehensible, that would be awesome :smile:

File: doc/Witchcraft.Monoid.html [hunk 2]
```diff
@@ -115,7 +116,7 @@ <h2 class="sidebar-projectVersion">
 <p>Integer division is not a monoid.
 Because you cannot divide by zero, the property does not hold for all values in the set.</p>
 <h1>Notes</h1>
-<p>You can of course use abuse this protocol to define a fake “monoid” that behaves differently.
+<p>You can of course use abuse this protocol to define a fake ‘monoid’ that behaves differently.
```
Comment (jenncoop) [inline doc/Witchcraft.Monoid.html]: I think you mean to remove the "use" here, no?

File: doc/Witchcraft.Monoid.html [hunk 3]
```diff
@@ -97,16 +97,17 @@ <h2 class="sidebar-projectVersion">
 <h1>Examples</h1>
 <h2>Theory</h2>
 <pre><code class="elixir"># Pseudocode
-id = 0
+identity = 0
 op = &amp;(&amp;1 + &amp;2) # Integer addition
 op(34, id) == 34</code></pre>
```
Comment (jenncoop) [inline doc/Witchcraft.Monoid.html]: I might be misunderstanding, but `id` should be `identity` here, no?

File: doc/Witchcraft.Monoid.html [hunk 4]
```diff
@@ -151,7 +152,7 @@ <h1 class="section-heading">
     </h2>
     <div class="summary-row">
   <div class="summary-signature">
-    <a href="#id/1">id(a)</a>
+    <a href="#identity/1">identity(a)</a>
   </div>
   
     <div class="summary-synopsis"><p>Get the idenity (‘zero’) element of the monoid by passing in any element of the set</p>
```
Comment (jenncoop) [inline doc/Witchcraft.Monoid.html]: Typo: "idenity"

File: doc/extra-api-reference.html [hunk 5]
```diff
@@ -137,7 +137,9 @@ <h1 class="section-heading">Modules</h1>
 <div class="summary-row">
   <div class="summary-signature"><a href="Witchcraft.Monoid.Properties.html">Witchcraft.Monoid.Properties</a></div>
   
-    <div class="summary-synopsis"><p>Check samples to confirm that your data adheres to monoidal properties</p>
+    <div class="summary-synopsis"><p>Check samples of your monoid to confirm that your data adheres to the
+monoidal properties. <em>All members</em> of your datatype should adhere to these rules.
+There are placed here as a quick way to spotcheck some of your values</p>
```
Comment (jenncoop) [inline doc/extra-api-reference.html]: "They" are placed here...?

File: lib/witchcraft.ex (hunk 1 above)
Comment (jenncoop) [inline lib/witchcraft.ex]: kk :+1:

**Comment by expede** [comment]
Awesome proofreading, @jenncoop! :smile:

**Comment by expede** [comment]
Hey @jenncoop, I've had to branch from this PR. Am I good to merge this into `master`?

Don't worry about CI failing; CircleCI doesn't support Elixir out of the box (I need to configure it, there's a story). They pass locally (run `mix test` from the project root).

**Comment by jenncoop** [comment]
cath5

david-hasselhoff-as-michael-knight-in-knightrider-thumbs-up

onward_human

**Pull request merged**